src/app/project/project-documents/application-sort/application-sort.component.ts

```diff
@@ -0,0 +1,243 @@
+      width: '15%'
+    },
+    {
+      name: 'Type',
+      value: 'type',
+      width: '15%'
+    },
+    {
+      name: 'Milestone',
+      value: 'milestone',
+      width: '15%'
+    },
+    {
+      name: 'Legislation',
+      value: 'legislation',
+      width: '10%'
+    }
+  ];
+
+
+  constructor(
+    private route: ActivatedRoute,
+    private router: Router,
+    public api: ApiService,
+    private _changeDetectionRef: ChangeDetectorRef,
+    private storageService: StorageService,
+    private searchService: SearchService,
+    private snackBar: MatSnackBar,
+    private utils: Utils,
+    private documentService: DocumentService,
+    private tableTemplateUtils: TableTemplateUtils,
+    private configService: ConfigService
+  ) {
+  }
+
+  ngOnInit() {
+    this.currentProject = this.storageService.state.currentProject.data;
+    this.storageService.state.editedDocs = [];
+
+    this.route.params
+      .takeUntil(this.ngUnsubscribe)
+      .subscribe(params => {
+        this.tableParams = this.tableTemplateUtils.getParamsFromUrl(params, null, 10);
+        if (this.tableParams.sortBy === '') {
+          this.tableParams.sortBy = '+sortOrder,-datePosted,+displayName';
+          this.tableTemplateUtils.updateUrl(this.tableParams.sortBy, this.tableParams.currentPage, this.tableParams.pageSize, null, this.tableParams.keywords);
+        }
+        this._changeDetectionRef.detectChanges();
+      });
+
+    this.route.data
+      .takeUntil(this.ngUnsubscribe)
+      .subscribe((res: any) => {
+        if (res) {
+          if (res.documents && res.documents[0].data.meta && res.documents[0].data.meta.length > 0) {
+            this.tableParams.totalListItems = res.documents[0].data.meta[0].searchResultsTotal;
+            this.documents = res.documents[0].data.searchResults;
+          } else {
+            this.tableParams.totalListItems = 0;
+            this.documents = [];
+          }
+          this.setRowData();
+          this.loading = false;
+          this._changeDetectionRef.detectChanges();
+        } else {
+          this.openSnackBar('Uh-oh, couldn\'t load documents', 'Close');
+          this.router.navigate(['/search']);
+        }
+      });
+  }
+
+  setRowData() {
+    let list = [];
+    if (this.documents && this.documents.length > 0) {
+      this.documents.forEach((item: any) => {
+        list.push(item);
+      });
+      this.tableData = new TableObject(
+        ApplicationSortTableRowsComponent,
+        list,
+        this.tableParams
+      );
+    }
+  }
+
+  updateSortOrder(document) {
+    console.log(document);
+  }
+
+  onSave() {
+    let formData = new FormData();
+    this.storageService.state.editedDocs.forEach((document: any) =>  {
+      console.log(document.sortOrder);
+      // document service put id and sort order
+      formData.set('sortOrder', document.sortOrder);
+      this.documentService.update( formData, document._id )
+      .takeUntil(this.ngUnsubscribe)
```

Comment:
Are these two functions required if we are not doing anything with the result in `subscribe`?

Reply:
for some reason when I don't have the subscribe it doesn't save. I think because this.documentService.update() returns an api call that you have to subscribe to?